server.js

```diff
@@ -1,38 +1,130 @@
-import express from 'express'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
+import express from "express";
+import cors from "cors";
+import mongoose from "mongoose";
+import listEndpoints from "express-list-endpoints";
+
+import booksData from "./data/books.json";
+
+const mongoUrl = process.env.MONGO_URL || "MONGO_URL";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(express.json())
+app.use(cors());
+app.use(express.json());
+
+const Book = mongoose.model("Book", {
+  bookID: Number,
+  title: String,
+  authors: String,
+  averageRating: Number,
+  isbn: Number,
+  isbn13: Number,
+  languageCode: String,
+  numPages: Number,
+  ratingsCount: Number,
+  textReviewsCount: Number,
+});
+
+if (process.env.RESET_DATABASE) {
+  const seedDatabase = async () => {
+    await Book.deleteMany({});
+
+    booksData.forEach((item) => {
+      const newBook = new Book(item);
+      newBook.save();
+    });
+  };
+  seedDatabase();
+}
+
+// Our own middleware that checks if the database is connected before going to our endpoints
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res.status(503).json({
+      response: "Service unavailable",
+      succes: false,
+    });
+  }
+});
+
+// Gives you a list of all the endpoints
+app.get("/", (req, res) => {
+  res.send(listEndpoints(app));
+});
+
+// Get all the books
+app.get("/books", async (req, res) => {
+  let books = await Book.find(req.query);
+
+  if (req.query.numPages) {
+    const booksByPages = await Book.find().gt("numPages", req.query.numPages);
+    books = booksByPages;
+  }
+  res.json(books);
```

Comment:
The response here is different if you compare to the response from line 78-81. I think it's better to have a consistent way so it's easier for the frontend to understand this API. Otherwise they have to code differently when they want to retrieve data from each response. For example on line 79 you need to fetch the data with a .response but not here on this response.